src/lib/_imports/elements/html-elements.cms.css

```diff
@@ -111,6 +111,10 @@ Styleguide Elements.TextContent
   /* https://dev.to/alvaromontoro/the-disappearing-line-a-css-mystery-3e35 */
   width: 100%;
 }
+/* HELP: What is the right way to fix this? See comment above `width: 100%;` */
+& .blog-content hr {
+  width: unset;
+}
```

Comment:
**No, really… Help.**

Should I:

**A. Do it "Right"**
1. Figure out where uses an `hr` as a flex or grid item.
2. Move the `width: 100%` to that use case

**B. Do it "Less Wrong"**
1. Move this `hr` style to the news/blog CSS.
2. Know that any other `<hr>` near floated content will need this arcane fix also.

**C. Do it "The Obvious Way I Don't See"**
1. You tell me.

Reply:
Because [`border-top` is used](https://github.com/TACC/Core-Styles/pull/310/files/830e93c7f6c37e4a264068825945e5231c25c2a8#diff-0b9db2def0d27b9bda457da24321dc12e4714a43082580f3bfee0dd21bae34f4R101-R113) on the `<hr>`[^1] tag. If `border-bottom` offers a technical advantage, do share.

I agree. (A) is how I should have done and should now do this. I think I just needed a kick in the kiester.

[^1]: The `<hr>` tag is used at all because it is used appropriately for a horizontal rule (and is appropriately available in the WYSIWYG editor). To add a border-bottom on another element would — I think — require a class on a `<p>` or some CSS specific to each article.

Reply:
**A.1.** [The `width: 100%` is to make visible any `<hr>` in a flex container.](https://github.com/TACC/Core-Styles/pull/310/files/830e93c7f6c37e4a264068825945e5231c25c2a8#diff-0b9db2def0d27b9bda457da24321dc12e4714a43082580f3bfee0dd21bae34f4R111-R113) And I think there are **zero** such instances. **I intend to delete the `width:100%` code.**  _If there are, I welcome an `<hr>` to disappear, us receive a complaint, and me fix it._

<details>

The `hr` tags in the Core-Styles, Core-CMS, and TUP-UI codebases are **not** within a flex (nor grid) container. History tells me it was added in TUP CMS during development for undocumented use case.[^1] Bootstrap 4 columns are **not** flex containers. Bootstrap 4 rows **are, but** an `<hr>` is not able to be added between columns (via Django CMS) nor would H.P. (CMS content admin) hack that in, nor would I allow it (we worked together closely when this code was created). The TUP-CMS News content is the only place I imagine uses `<hr>` and flex, and the news content wrapper is **not** a flex container.

</details>

[^1]: (1) [Added to TUP.](https://github.com/TACC/tup-ui/pull/174/files#diff-6551d51b40a593f67415531bd61a41ca09dd0a7d3f4965c648c417c27e94a698) (2) [Removed form TUP.](https://github.com/TACC/tup-ui/pull/202/files#diff-6551d51b40a593f67415531bd61a41ca09dd0a7d3f4965c648c417c27e94a698) (3) [Core-Styles receives it during migration.](https://github.com/TACC/Core-Styles/commit/ad0b5c3) (4) [Core-CMS installs Core-Styles that has it.](https://github.com/TACC/Core-CMS/pull/620/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519)

Reply:
Check out the new diff.

Reply:
@R-Tomas-Gonzalez Check out the new diff. 🙂

Reply:
@R-Tomas-Gonzalez and I synced. The diff as it is now (remove `width: 100%`) is what should be done for Core-Styles. But to fix the upstream bug, Core-CMS needs `overflow: hidden` for `<hr>` in blog.[^1]

[^1]: Don't add it here, because though it seems benign to put `overflow: hidden` on every `<hr>`, it might not be. (It seemed benign to `width: 100%` on every `<hr>`, and then this bug began. 😉)